plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryPageSink.java

```diff
@@ -73,16 +81,21 @@ public BigQueryPageSink(
         this.pageSinkIdColumnName = requireNonNull(pageSinkIdColumnName, "pageSinkIdColumnName is null");
         checkArgument(temporaryTableName.isPresent() == pageSinkIdColumnName.isPresent(),
                 "temporaryTableName.isPresent is not equal to pageSinkIdColumn.isPresent");
-        TableName tableName = temporaryTableName
+        tableName = temporaryTableName
                 .map(table -> TableName.of(remoteTableName.getProjectId(), remoteTableName.getDatasetName(), table))
                 .orElseGet(remoteTableName::toTableName);
-        // TODO: Consider using PENDING mode
-        WriteStream stream = WriteStream.newBuilder().setType(COMMITTED).build();
+
         CreateWriteStreamRequest createWriteStreamRequest = CreateWriteStreamRequest.newBuilder()
                 .setParent(tableName.toString())
-                .setWriteStream(stream)
+                .setWriteStream(WriteStream.newBuilder().setType(PENDING).build())
                 .build();
-        this.writeStream = client.createWriteStream(createWriteStreamRequest);
+        writeStream = client.createWriteStream(createWriteStreamRequest);
+        try {
+            streamWriter = JsonStreamWriter.newBuilder(writeStream.getName(), writeStream.getTableSchema(), client).build();
+        }
+        catch (DescriptorValidationException | IOException | InterruptedException e) {
+            throw new TrinoException(BIGQUERY_BAD_WRITE, "Failed to create stream writer", e);
```

Comment:
include original message in the final message (`firstNonNull(e.getMessage(), e)`)

oftentimes, the exc message is all user gets, not full stacktrace